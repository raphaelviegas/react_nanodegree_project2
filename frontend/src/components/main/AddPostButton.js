import React from 'react'
import { Link } from 'react-router-dom'
import MaterialIcon from 'material-icons-react'

export default () => {
  return (
    <div>
      <div className="fixed-action-btn">
        <Link className="btn-floating btn-large cyan darken-4" to="/addPost">
          <MaterialIcon icon="playlist_add" invert />
        </Link>
      </div>
    </div>
  )
}