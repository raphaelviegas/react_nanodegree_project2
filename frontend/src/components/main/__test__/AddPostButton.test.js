import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'

import AddPostButton from '../AddPostButton'
import { Link } from 'react-router-dom'
import MaterialIcon from 'material-icons-react'

describe('<AddPostButton />', () => {
    let wrapper
    beforeEach(() => {
        wrapper = shallow(<AddPostButton />)
    })

    it('Should render one link tag to /addPost', () => {
        const auxiliarIcon = "playlist_add" //Used because material Icons shows an alert if the prop icon is undefined
        expect(wrapper.containsMatchingElement(<Link to="/addPost"><MaterialIcon icon={auxiliarIcon} /></Link>)).toEqual(true)
    })

    it('Should render correctly', () => {
        const tree = renderer.create(
            <MemoryRouter initialEntries={[{ pathname: '/', key: 'testKey' }]}>
                <AddPostButton />
            </MemoryRouter>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

})
