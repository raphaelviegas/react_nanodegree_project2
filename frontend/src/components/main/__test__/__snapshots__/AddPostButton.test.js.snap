// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddPostButton /> Should render correctly 1`] = `
<div>
  <div
    className="fixed-action-btn"
  >
    <a
      className="btn-floating btn-large cyan darken-4"
      href="/addPost"
      onClick={[Function]}
    >
      <i
        className="material-icons md-24 md-light "
        style={
          Object {
            "color": "",
            "fontSize": "md-24",
          }
        }
      >
        playlist_add
      </i>
    </a>
  </div>
</div>
`;
